presto-orc/src/test/java/com/facebook/presto/orc/TestOrcRecordReader.java

```diff
@@ -0,0 +1,143 @@
+                                "col_3_struct"
+                        )),
+                // 1
+                new OrcType(OrcType.OrcTypeKind.INT,
+                        ImmutableList.of(),
+                        ImmutableList.of()),
+                // 2
+                new OrcType(OrcType.OrcTypeKind.MAP,
+                        ImmutableList.of(4, 5),
+                        ImmutableList.of(
+                                "col_0_string",
+                                "col_1_date"
+                        )),
+                // 3
+                new OrcType(OrcType.OrcTypeKind.LIST,
+                        ImmutableList.of(6),
+                        ImmutableList.of("col_0_double")),
+                // 4
+                new OrcType(OrcType.OrcTypeKind.STRING,
+                        ImmutableList.of(),
+                        ImmutableList.of()),
+                // 5
+                new OrcType(OrcType.OrcTypeKind.DATE,
+                        ImmutableList.of(),
+                        ImmutableList.of()),
+                // 6
+                new OrcType(OrcType.OrcTypeKind.DOUBLE,
+                        ImmutableList.of(),
+                        ImmutableList.of()),
+                // 7
+                new OrcType(OrcType.OrcTypeKind.STRUCT,
+                        ImmutableList.of(8, 9),
+                        ImmutableList.of(
+                                "nested_col_0_double",
+                                "nested_col_1_boolean"
+                        )),
+                // 8
+                new OrcType(OrcType.OrcTypeKind.DOUBLE,
+                        ImmutableList.of(),
+                        ImmutableList.of()),
+                // 9
+                new OrcType(OrcType.OrcTypeKind.BOOLEAN,
+                        ImmutableList.of(),
+                        ImmutableList.of())
+        );
+        final Map<Integer, Type> includedColumnsGood1 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE)
+        );
+        final Map<Integer, Type> includedColumnsGood2 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE),
+                3, new RowType(ImmutableList.of(DoubleType.DOUBLE, BooleanType.BOOLEAN), Optional.of(ImmutableList.of("nested_col_0", "nested_col_1")))
+        );
+        final Map<Integer, Type> includedColumnsBad1 = ImmutableMap.of(
+                0, BooleanType.BOOLEAN,
+                1, new MapType(VarcharType.createVarcharType(2), DateType.DATE),
+                2, new ArrayType(DoubleType.DOUBLE)
+        );
+        final Map<Integer, Type> includedColumnsBad2 = ImmutableMap.of(
+                0, BigintType.BIGINT,
+                1, new ArrayType(DoubleType.DOUBLE),
```

Comment:
`1, new ArrayType(DoubleType.DOUBLE), // should be map<varchar,date>`
Same for the other bad ones.